### PR TITLE
104 add blog posts

### LINKED ISSUE #104: Preserving blogs.
Consider the need to preserve the blogs, how to preserve existing ones and maintain links+redirects.

The need: have the old blog content available at the same URLs as before

We could:
- take the ghost dump and replicate the files in the gatsby repo.
  - Create a markdown file per post
- Download images from server
- Render the markdown using components we have already built for the site
  - https://www.npmjs.com/package/markdown-to-jsx#optionsoverrides---override-any-html-tags-representation
- Make sure the slugs match the path off the root of the site so cached paths work still
- extract meta data from ghost blog page so search results and linking works correctly.

Acceptance Criteria:
- [x] The blogs are stored as markdown in the repo
- [x] A script exists for converting a ghost dump into gatsby blog posts
- [x] Images are resolved correctly
- [x] We have redirects configured for blogs so that existing urls work

### PR DESCRIPTION
**Pull request by stphnnnn**
Fixes #104

example blog post can be found at `/blog/smart-sass-dynamic-sass-variables-with-php-smarty`

Todo:

- [x] add static images
- [x] lists page
- [x] add authors
- [x] add redirects

### CODE REVIEW

File: generate-blog-posts.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const fs = require('fs')
+const data = require('./src/data/blog/posts.json')
+
+const { posts } = data.db[0].data
+
+// Converts "null" to null, "false" to false etc. Everything is returned as a string wrapped in "s
+const convertType = value => {
+  try {
+    return new Function('return ' + value + ';')()
```
Comment (RpprRoger) [inline generate-blog-posts.js]: So this feels `eval`. Can't we use JSON.parse instead?
```js
JSON.parse('null') === null
JSON.parse('true') === true
```
```suggestion
    return JSON.parse(value)
```

File: gatsby-browser.js [hunk 2]
```diff
@@ -4,4 +4,4 @@
  * See: https://www.gatsbyjs.org/docs/browser-apis/
  */
 
-// You can delete this file if you're not using it
+require('./prism-theme.css')
```
Comment (RpprRoger) [inline gatsby-browser.js]: Rather can downloading and committing this file can we use a module like [`prism-themes`](https://www.npmjs.com/package/prism-themes) and instead `require('prism-themes/themes/prism-cb.css')`

File: src/components/Heading.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { fontSizes } from '../theme'
 
-const Heading = ({ level, children, size }) => {
+const Heading = ({ level = 1, children, size = 'large' }) => {
```
Comment (RpprRoger) [inline src/components/Heading.js]: I don't think there's a logical default for level and size, in the place that we define the replacements we can manually create them.
```suggestion
const Heading = ({ level, children, size }) => {
```

File: src/templates/blog-post.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import Heading from '../components/Heading'
+import { ExternalLink } from '../components/Link'
+import VerticalSpacing from '../components/VerticalSpacing'
+import PageMeta from '../components/PageMeta'
+
+const renderAst = new rehypeReact({
+  createElement: React.createElement,
+  components: {
+    h1: Heading,
+    h2: Heading,
```
Comment (RpprRoger) [inline src/templates/blog-post.js]:
```suggestion
    h2: props => (<Heading {...props} level={2} size="medium" />),
```

File: src/templates/blog-post.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import Paragraph from '../components/Paragraph'
+import Heading from '../components/Heading'
+import { ExternalLink } from '../components/Link'
+import VerticalSpacing from '../components/VerticalSpacing'
+import PageMeta from '../components/PageMeta'
+
+const renderAst = new rehypeReact({
+  createElement: React.createElement,
+  components: {
+    h1: Heading,
```
Comment (RpprRoger) [inline src/templates/blog-post.js]:
```suggestion
    h2: props => (<Heading {...props} level={1} size="large" />),
```

**Comment by RpprRoger** [comment]
Thanks @stphnnnn!

File: gatsby-browser.js (hunk 2 above)
Comment (stphnnnn) [reply gatsby-browser.js]: this is so we can add our own custom styles to the code blocks :+1:

File: package.json [hunk 6]
```diff
@@ -3,19 +3,26 @@
...
     "gatsby": "2.9.0",
     "gatsby-image": "2.1.3",
     "gatsby-plugin-manifest": "2.1.1",
     "gatsby-plugin-offline": "2.1.1",
     "gatsby-plugin-react-helmet": "3.0.12",
-    "gatsby-plugin-sharp": "2.1.3",
+    "gatsby-plugin-sharp": "^2.2.1",
     "gatsby-plugin-styled-jsx": "3.0.5",
-    "gatsby-source-filesystem": "2.0.39",
+    "gatsby-remark-copy-linked-files": "^2.1.0",
```
Comment (RpprRoger) [inline package.json]: Can we use exact versions please

File: src/components/WorkDescription/index.js [hunk 7]
```diff
@@ -43,11 +43,13 @@ const WorkDescription = ({
...
-              <T heavy>{title}</T>
+            <Heading level={level}>
+              <T size="medium" heavy>
+                {title}
+              </T>
             </Heading>
-            <Heading level={level + 1} size="normal">
-              <T>{subtitle}</T>
+            <Heading level={level + 1}>
+              <T size="normal">{subtitle}</T>
```
Comment (HedgehogHolly) [inline src/components/WorkDescription/index.js]: isn't text normal as standard?

**Pull request merged**